packages/cli/src/lib/new/utils.ts

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright 2024 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { Template } from './types';
+
+export interface Options extends Record<string, string | boolean> {
+  id: string;
+  private: boolean;
+  baseVersion: string;
+  license: string;
+  targetPath: string;
+  owner: string;
+  scope: string;
+  moduleId: string;
+}
+
+export const resolvePackageName = (options: {
+  baseName: string;
+  scope?: string;
+  plugin: boolean;
+}) => {
+  const { baseName, scope, plugin } = options;
+  if (scope) {
+    const scopeSanitized = scope.replace(/[@/]/g, '');
+    if (plugin) {
+      const pluginPrefix = scopeSanitized.startsWith('backstage')
+        ? 'plugin'
+        : 'backstage-plugin';
+      return `@${scopeSanitized}/${pluginPrefix}-${baseName}`;
+    }
+    return `@${scopeSanitized}/${baseName}`;
+  }
+
+  return plugin ? `backstage-plugin-${baseName}` : baseName;
+};
+
+export function populateOptions(
+  prompts: Record<string, string>,
+  template: Template,
+): Options {
+  return {
+    id: prompts.id ?? '',
+    private: !!prompts.private,
+    baseVersion: prompts.baseVersion ?? '0.1.0',
+    owner: prompts.owner ?? '',
```

Comment:
Thinking it's better to keep this as optional rather than default to `''`, or is it more convenient to give it a value?